### ISSUE TITLE
Combat screen hit / death explosion effects

### ISSUE THREAD

**Issue by tewey** [1oom-fork/1oom#1]
Build: 1.6.2
Bug / unexpected behavior:
- Hitting a stack and not wiping out the last ship plays the "big explosion" animation.
- Hitting a stack, destroying the last ship thus wiping out the stack plays the "small explosion" animation. 

It seems that this is the reverse of the behavior in Master of Orion 1.3. Tested on both the DOS and the SDL2 executables with default settings / New Game and the behavior is the same.

**Comment by 1oom-fork** [comment]
The second case is actually a scaled "big explosion" so we need to better understand the expected behavior.
ui/classic/uibattle.c(1161):
si = (b->hp1 > 0) ? b->hp1 * 3 : 1;
if ((b->num > 0) || (si < damage)) {
    v4 = si;
} else {
    v4 = damage; /* & 0xffff */
}
if ((si / v4) >= 3) {
    v4 = si / 3 + 1;
}
scale = (v4 * 100) / si;

**Comment by tewey** [comment]
Interesting, the scaling bit was news for me. I did some more manual testing and this is what I found out:

MoO 1.3:
- Non-stack-wiping damage seems to scale the explosion, thus often displaying a small one unless dealing heavy damage.
- Stack-wiping damage always displays the big explosion regardless of the damage dealt.

1oom-fork:
- Non-stack-wiping damage always displays the big explosion regardless of the damage dealt.
- Stack-wiping damage seems to scale the explosion, thus often displaying a small one unless dealing heavy damage.

Essentially the cases are reversed based on what I can gather from this admittedly quick manual testing. So my guess is the scaling is applied in the wrong case.

This got me thinking that the different scales of explosion in MoO were probably put in place to convey useful information to the player. During huge fleet battles it's easier to see at glance when a connecting hit is particularly nasty.
